djgeojson/views.py

```diff
@@ -88,7 +88,32 @@ def get_queryset(self):
         """
         Inspired by Glen Roberton's django-geojson-tiles view
         """
-        self.z, self.x, self.y = map(int, self.args[:3])
+        try:
+            self.z, self.x, self.y = map(int, self.args[:3])
+        except AttributeError:
+            # let's try to get these as kwargs
+
+            self.z = self.kwargs.get('z', None)
+            self.x = self.kwargs.get('x', None)
+            self.y = self.kwargs.get('y', None)
+
+            if self.z is None:
+                raise ValueError('View parameter Z not provided in URL.')
+
+            if self.x is None:
+                raise ValueError('View parameter X not provided in URL.')
+
+            if self.y is None:
+                raise ValueError('View parameter Y not provided in URL.')
+
+            try:
+                self.z = int(self.z)
+                self.x = int(self.x)
+                self.y = int(self.y)
+            except ValueError:
```

Comment:
You could group the whole code to make it shorter (*since int(None) triggers TypeError*)

```
try:
    self.z = int(self.kwargs.get('z'))
    ...
except (ValueError, TypeError):
     ...

```